### DISCUSSION TITLE
Can't get cy.intercept to add custom headers

### DISCUSSION THREAD

**Discussion by mzedeler** [cypress-io/cypress#14989] [Questions and Help] [answered]
I am using Cypress 6.4, and like the documentation states, I am trying to add headers to all outgoing requests (in this case, to switch off caching), and it doesn't work. Before any other calls to `cy.intecept`, I do this:

```
      cy.intercept(/.*/, req => {
        req.headers['cache-control'] = 'no-cache'
        delete req.headers['if-none-match']
      })
```

If I look in the network tab, the requests are not modified. Also, I can see that the request rewriting block above is actually called on every outgoing request.

**Comment by jennifer-shehane** [comment, marked as answer]
The header modifications are actually working, they just won’t show in the browser’s Network tab, because the browser logs the request before Cypress intercepts and modifies it. Confirm it with `cy.wait()` instead:

```js
cy.intercept({ url: '**/*', middleware: true }, (req) => {
  delete req.headers['if-none-match']
}).as('reqs')

cy.wait('@reqs').its('request.headers').should('not.have.property', 'if-none-match')
```

For your caching goal specifically, overriding `cache-control` on the response is more reliable, and note that requests already served from the browser cache never hit the network layer so the intercept won’t fire at all. Both are documented here:

- https://on.cypress.io/intercept#Add-modify-or-delete-a-header-to-all-outgoing-requests
- https://on.cypress.io/intercept#cyintercept-and-request-caching
